compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -987,6 +1020,55 @@ impl<'context> Elaborator<'context> {
         self.generics.clear();
     }
 
+    fn check_parent_traits_are_implemented(&mut self, trait_impl: &UnresolvedTraitImpl) {
+        let Some(trait_id) = trait_impl.trait_id else {
+            return;
+        };
+
+        let Some(object_type) = &trait_impl.resolved_object_type else {
+            return;
+        };
+
+        let Some(the_trait) = self.interner.try_get_trait(trait_id) else {
+            return;
+        };
+
+        // Note: we only check if the immediate parents are implemented, we don't check recursively.
+        // Why? If a parent isn't implemented, we get an error. If a parent is implemented, we'll
+        // do the same check for the parent, so this trait's parents parents will be checked, so the
+        // recursion is guaranteed.
+        for parent_trait_bound in the_trait.trait_bounds.clone() {
+            let Some(parent_trait) = self.interner.try_get_trait(parent_trait_bound.trait_id)
+            else {
+                continue;
+            };
+
+            if self
+                .interner
+                .try_lookup_trait_implementation(
+                    object_type,
+                    parent_trait_bound.trait_id,
+                    &parent_trait_bound.trait_generics.ordered,
+                    &parent_trait_bound.trait_generics.named,
+                )
+                .is_err()
+            {
+                let mut the_trait = parent_trait.name.to_string();
+                if !parent_trait_bound.trait_generics.is_empty() {
+                    the_trait.push('<');
+                    the_trait.push_str(&parent_trait_bound.trait_generics.to_string());
+                    the_trait.push('>');
+                }
```

Comment:
trait_generics will already do this formatting when converting it to a string, no need for the extra is_empty or <>